File: custom_msg_pkg/src/custom_msg_pub_node.py
#!/usr/bin/env python3
#conding: utf-8
import rclpy
from custom_msgs.msg import CustomMsg
import time

pub = None

def timer_callback():
    global pub

    msg = CustomMsg()

    msg.grade = "A+"
    msg.star = 50
    msg.data = "Hello, I send custom msg data!"
    pub.publish(msg)

def main(args=None):
    global pub

    rclpy.init(args=None)
    node = rclpy.create_node('custom_msg_pub_node')

    print("Hello World, I am the custom msg pub node! [Python]\n")

    pub = node.create_publisher(CustomMsg, 'custom_msg_topic', 10)

    node.create_timer(1.0, timer_callback)

    rclpy.spin(node)

    node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()

File: custom_msg_pkg/src/custom_msg_pub_node.cpp
#include <rclcpp/rclcpp.hpp>
#include <custom_msgs/msg/custom_msg.hpp>

int main(int argc, char const *argv[])
{
    rclcpp::init(argc, argv);


    printf("Hello World, I am the custom msg pub node!\n");

    rclcpp::Node::SharedPtr node = rclcpp::Node::make_shared("custom_msg_pub_node");

    rclcpp::Publisher<custom_msgs::msg::CustomMsg>::SharedPtr pub = node->create_publisher<custom_msgs::msg::CustomMsg>(
        "custom_msg_topic",
        10
    );

    custom_msgs::msg::CustomMsg msg;

    msg.grade = "A+";
    msg.star = 50;
    msg.data = "Hello, I send custom msg data!";

    rclcpp::Rate loop_rate(1);

    while(rclcpp::ok())
    {
        pub->publish(msg);

        loop_rate.sleep();
    }

    return 0;
}
